adapters/mock/adapter.ts

```diff
@@ -48,6 +48,11 @@ export async function mockFetch({
   });
 
   const next = mockTestRequests.responseList.shift()!;
+  if (!next) {
```

Comment:
This was failing silently with a `reading .body from undefined` error that was super confusing, so I added a more explicit error here.